### DISCUSSION TITLE
Pull_request_target checkout details?

### DISCUSSION THREAD

**Discussion by kingthorin** [community/community#25721] [Actions]
<p>I have a few workflows that lint or otherwise check PRs then comment. Currently I’m using a step:</p>
<pre><code class="lang-yaml">    - name: Checkout
      uses: actions/checkout@v2
      with:
        ref: ${{github.event.pull_request.head.ref}}
        repository: ${{github.event.pull_request.head.repo.full_name}}
</code></pre>
<p>But I’m wondering if I could/should just change to <code>fetch-depth: 0</code> instead of specifying the <code>with</code> params?</p>
<p>I don’t have a good justification for why I think this should work out (I haven’t tested) but setting fetch depth 0 on other totally different workflows (ci builds etc) seems to properly checkout/target the PR content.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Depends on what you want the linter to work with. <code>fetch-depth: 0</code> only makes sense if you want to look at the history, not just the latest code.</p>
<p>By default (using <code>github.sha</code>) the checked out commit for a PR workflow will be a temporary merge that represents what would be the result of the merge, not the latest commit in the PR. I think that makes sense for a lot of cases (e.g. I want to know if tests would pass after merge), but if you want to work on the branch itself checking out the head branch is the right approach.</p>

**Comment by kingthorin** [comment]
<p>Okay that makes sense, when I had originally been setting them up I found that the default checkout wouldn’t let me comment on the PR due to permissions. Since it’s a temporary merge (of what will be), that makes sense.</p>
<p>Thanks <a class="mention" href="/airtower-luna">@airtower-luna</a> !</p>
